waiter/src/waiter/async_request.clj

```diff
@@ -172,4 +173,5 @@
     (let [param-map {:host host, :location location, :port port, :request-id request-id, :router-id router-id, :service-id service-id}
           status-location (route-params->uri "/waiter-async/status/" param-map)]
       (log/info "updating status location to" status-location "from" location)
-      (swap! response-headers assoc "location" status-location))))
\ No newline at end of file
+      (-> response
+          (assoc-in [:headers "location"] status-location)))))
```

Comment:
Do we need to worry about the case where `location` might already be in headers but in a different case?

Reply:
Before this PRl, we were overriding existing headers when we `merge`d over headers with `@response-headers`.  This function is only called for `202` responses where `Location` starts with `/` (ensuring these conditions is handled in `inspect-for-202-async-request-response`.

Reply:
okay, `inspect-for-202-async-request-response` also uses lower-case `location`.